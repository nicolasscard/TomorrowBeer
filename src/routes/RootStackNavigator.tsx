import React, { useContext } from 'react';
import { View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import { BeerListScreen, BeerDetailScreen } from '../screens';
import { ThemeContext } from '../context/themeContext/ThemeContext';

export type StackParams = {
  BeerList: undefined,
  BeerDetail: {id: number},
}

const Stack = createStackNavigator<StackParams>();

export const RootStackNavigator = () => {
  const { theme } = useContext( ThemeContext );

  return (
    <View style={{ flex: 1, backgroundColor: theme.colors.background }}> 
      <NavigationContainer theme={ theme }>
        <Stack.Navigator initialRouteName="BeerList">
          <Stack.Screen name="BeerList" options={{ headerShown: false }} component={ BeerListScreen } />
          <Stack.Screen name="BeerDetail" options={{ headerShown: false }} component={ BeerDetailScreen } />
        </Stack.Navigator>
      </NavigationContainer>
    </View>
  );
}
