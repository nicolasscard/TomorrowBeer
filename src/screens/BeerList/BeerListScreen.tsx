import React, { useEffect } from 'react';
import { Text, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { usePunkApi } from '../../hooks/usePunkApi';
import { FlatList } from 'react-native-gesture-handler';
import { Beer } from '../../interfaces/punkApiResponse';

const BeerListScreen = () => {
  const { beers, isLoading, loadBeers } = usePunkApi();

  useEffect(() => {
    loadBeers();
  }, []);

  const renderItem = (item: Beer) => {
    return (
      <View>
        <Text>{ item.name }</Text>
      </View>
    );
  };
  
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <Text style={{ fontWeight: 'bold', paddingBottom: 20 }}>BeerListScreen</Text>
      {beers.length > 0 &&
        <FlatList
            data={beers}
            renderItem={({item}) => renderItem(item)}
            keyExtractor={(item) => item.id.toString()}
            refreshing={isLoading}
        />
      }
    </SafeAreaView>
  );
}

export default BeerListScreen;
