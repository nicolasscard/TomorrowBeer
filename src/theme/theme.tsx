import { ThemeState } from '../interfaces/themeState';

export const lightTheme: ThemeState = {
    dark: false,
    colors: {
        primary: '#9F3A52',
        background: 'white',
        card: 'white',
        text: 'black',
        border: '#e6e6e6',
        notification: 'black',
    },
}

export const darkTheme: ThemeState = {
    dark: false,
    colors: {
        primary: '#9F3A52',
        background: 'white',
        card: 'white',
        text: 'black',
        border: '#e6e6e6',
        notification: 'black',
    },
}
