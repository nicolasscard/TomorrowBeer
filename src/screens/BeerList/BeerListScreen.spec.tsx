import React from 'react';
import renderer from 'react-test-renderer';
import { StackNavigationProp } from '@react-navigation/stack';
import { RouteProp } from '@react-navigation/native';

import { StackParams } from '../../routes/RootStackNavigator';
import { lightTheme } from '../../theme/theme';
import BeerListScreen from './BeerListScreen';

type NavigationProps = StackNavigationProp<StackParams, 'BeerList', undefined>;
type PartialNavigationProps = Partial<NavigationProps>;

const mockThemeContextValue = { 
  theme: lightTheme,
  setDarkTheme: jest.fn(),
  setLightTheme: jest.fn(),
};

const mockUsePunkApiData = { 
  beers: [], 
  isLoading: false, 
  loadBeers: jest.fn() 
};

jest.mock('react', () => {
  const React = jest.requireActual('react');
  return {
    ...React,
    useContext: jest.spyOn(React, 'useContext'),
  }
});

jest.mock('../../context/themeContext/ThemeContext', () => ({
  ThemeContext: {
    Consumer: ({ children }) => children(mockThemeContextValue),
  },
}));

jest.mock('../../hooks/usePunkApi', () => ({
  usePunkApi: jest.fn(() => mockUsePunkApiData),
}));

describe('BeerList',  () => {
    it('should render correctly', () => {
      jest.spyOn(React, 'useContext').mockReturnValue(mockThemeContextValue);

      const mockNavigate: PartialNavigationProps = {
        navigate: jest.fn(),
      };
  
      const mockRoute: RouteProp<StackParams, 'BeerList'> = {
        key: '123',
        name: 'BeerList',
        params: undefined,
      };

      const component = renderer.create(
        <BeerListScreen navigation={mockNavigate as NavigationProps} route={mockRoute} /> 
      );
        
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  