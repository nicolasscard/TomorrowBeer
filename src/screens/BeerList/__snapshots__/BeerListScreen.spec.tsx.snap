// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerList should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      We did not find beers... =( 
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#9F3A52",
        "borderRadius": 56.25,
        "elevation": 6,
        "height": 112.5,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 75,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.27,
        "shadowRadius": 4.65,
        "top": 112.5,
        "width": 112.5,
      }
    }
  >
    <View
      style={
        {
          "position": "absolute",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 60,
            },
            undefined,
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó°ˆ²
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
